File: src/main/java/com/architecture_map/belarus/dto/ArchitecturalStyleDto.java
package com.architecture_map.belarus.dto;

import com.architecture_map.belarus.entity.ArchitecturalAttribute;
import com.architecture_map.belarus.entity.image.Image;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter(AccessLevel.PUBLIC)
@RequiredArgsConstructor
public class ArchitecturalStyleDto {

    private String name;
    private String description;
    private Set<ArchitecturalAttribute> architecturalAttributes;
    private String shortDescription;
    private String yearsActive;
    private Image demonstrativeImage;

}

File: src/main/kotlin/by/architecturemap/belarus/dto/ArchitecturalStyleDTO.kt
package by.architecturemap.belarus.dto

import by.architecturemap.belarus.entity.ArchitecturalAttribute
import by.architecturemap.belarus.entity.Image

data class ArchitecturalStyleDTO(
    var name: String? = null,
    var description: String? = null,
    var demonstrativeImage: Image? = null,
    var shortDescription: String? = null,
    var yearsActive: String? = null,
    var attributes: List<ArchitecturalAttribute> = listOf()
)
